ember-app/app/visitors/issue/references.js

```diff
@@ -0,0 +1,54 @@
+import Ember from 'ember';
+
+var IssueReferencesVisitor = Ember.Object.create({
+  visit: function(issue){
+    var promises = [
+      this.run(this.discoverIssues, issue),
+      this.run(this.discoverClosedIssues, issue)
+    ];
+
+    Ember.RSVP.all(promises).then((references)=> {
+      issue.set('issueReferences', _.flatten(references));
+    });
+  },
+
+  run: function(discovery, issue){
+    var _self = this;
+    return new Ember.RSVP.Promise((resolve, reject)=>{
+      resolve(discovery.call(_self, issue));
+      reject([]);
+    });
+  },
+
+  discoverIssues: function(issue){
+    var references = issue.get('cardRelationships')['issue-references'] || [];
+    var issuesById = issue.get('repo.board.issuesById');
+    return references.map((reference)=>{
+      if(issuesById.hasOwnProperty(reference.id)){
+        return issuesById[reference.id][0];
```

Comment:
Was thinking the need for `[0]` was less than ideal - found [`indexBy()`](http://underscorejs.org/#indexBy) for just this case:

>  Just like `groupBy`, but for when you know your keys are unique.